src/DesktopPixmap.cpp

```diff
@@ -0,0 +1,99 @@
+#include "DesktopPixmap.hpp"
+#include <iostream>
+#include <ostream>
+#include <xcb/xcb.h>
+#include <xcb/xproto.h>
+
+DesktopPixmap::DesktopPixmap (
+    const short x,          ///< x coordinate of the top left corner
+    const short y,          ///< y coordinate of the top left corner
+    const uint16_t height,  ///< height of display
+    const uint16_t width,   ///< width of display
+    const std::string &name ///< _NET_WM_NAME of display
+)
+{
+  this->x = x;
+  this->y = y;
+  this->height = height;
+  this->width = width;
+  this->name = name; // XXX Is this OK?
```

Comment:
I'm not sure that's ok. Name is a pointer itself(why, btw?), a pointer to pointer might lead to some unexpected errors.

Reply:
Looks like it does not matter for short strings.

As I won't change the passed `const` string, only move operations will be called by the compiler. But code generated with `-O3` flag looks the same both for the const reverence and by-value argument passes.